airflow/contrib/hooks/dingding_webhook_hook.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+
+from airflow import AirflowException
+from airflow.hooks.http_hook import HttpHook
+
+
+class DingdingWebhookHook(HttpHook):
+    """
+    This hook allows you to post messages to Dingding use webhook.
+    Dingding token get from conn_id extra field ``webhook_token`` for security.
+
+    For more message detail in
+    `Dingding custom bot <https://open-doc.dingtalk.com/microapp/serverapi2/qf2nxq>`_
+
+    :param http_conn_id: connection that has Dingding ``webhook_token`` in the extra field
+    :type http_conn_id: str
+    :param message_type: Message type you want to send to Dingding
+                         support text/link/markdown/actionCard/feedCard five types so far
+    :type message_type: str
+    :param message: The message you want to send on Dingding
+    :type message: str or dict
+    :param at_mobiles: Users you want to remind at this message.
+    :type at_mobiles: list of str
+    :param at_all: Should remind all people in group or not. If True, will overwrite ``at_mobiles``
+    :type at_all: bool
+    """
+
+    def __init__(self,
+                 http_conn_id='dingding_webhook_default',
+                 message_type='text',
+                 message=None,
+                 at_mobiles=False,
+                 at_all=False,
+                 *args,
+                 **kwargs
+                 ):
+        super(DingdingWebhookHook, self).__init__(*args, **kwargs)
+        self.http_conn_id = http_conn_id
+        self.message_type = message_type
+        self.message = message
+        self.at_mobiles = at_mobiles
+        self.at_all = at_all
+
```

Comment:
```suggestion
    self.base_url = "https://oapi.dingtalk.com"
    
```

This will then set the default host if the connection doesn't specify a host/path.

Reply:
I try some time. And find we could not just set `self.base_url = "https://oapi.dingtalk.com"` in here.
If `conn.host` is None, 
https://github.com/apache/airflow/blob/40acf579bcd8d2538a68e9bd86041048dbc94318/airflow/hooks/http_hook.py#L63
will raise error.
So I add describe in hook and `hwoto` use about we must add `host` to `dingding_default`.

Reply:
```
In [1]: from airflow.models.connection import Connection

In [2]: c = Connection()

In [3]: c.parse_from_uri("//:password@")

In [4]: c.host
Out[4]: ''

In [5]: c.host is None
Out[5]: False
```

Reply:
Oh but yes, I mis-read the code in http_hook. So my approach won't work.

Reply:
One option could be to just override `get_conn()` in our class to be hard-coded:

```
def get_conn(self):
   return requests.Session()
```

Since we already set base_url else where, and we don't want to support generic params (login/password, extra headers) this might actually be the best approach?

What do you think?

Reply:
That a good idea.

Reply:
I use https://github.com/apache/airflow/pull/4895/files#diff-0ddca7f4132e6c4a5dc1ee70d4008085R108 overwrite default value if user set specific value in connection `host`.